### ISSUE TITLE
Need ability to edit past posts

### ISSUE THREAD

**Issue by RobertSalm** [bluesky-social/social-app#6595]
### Describe the Feature

Users need the ability to edit past posts.  This alone would be a tremendous game changer to make BlueSky stand out amongst social networking services.

**Comment by Shohreh** [comment]
Since there's no (obvious) way to upvote: 👍

It's very annoying having to delete a post, possibly with pictures, just to fix a typo or a couple of words.

**Comment by RobertSalm** [comment]
I'd like to hear from the tech community how challenging it might be to code an edit feature, which I assume would be based on what can be edited (a la TikTok)--images, videos, text, and tags.  I suppose there's a philosophical reason other social media companies don't allow editing--probably their desire for posts to have an immediacy and permanence factor that they don't want altered?

**Comment by daffhunt** [comment]
A valid argument against allowing editing is that a malicious user could post something, wait for comments and reposts, and then alter it to make it look like people were commenting on something else entirely. Perhaps a compromise would be to allow editing the most recent post only, within a fairly short time window, just to deal with those "doh" moments when you spot the typo right after pressing Send. In any case, there should be a flag visible on edited messages and a way to view the original text

**Comment by surfdude29** [comment]
This is on the roadmap, please see #673 for discussion of this feature.

Also please see the **Record Versioning (Post Editing)** section of the 2024 Protocol Roadmap:

https://docs.bsky.app/blog/2024-protocol-roadmap#on-the-horizon

> **Record Versioning (Post Editing):** atproto already supports updating records in repositories: one example is updating bsky profile records. And preparations were made early in the protocol design to support post editing while avoiding misleading edits. Ideally, it would also be possible to (optionally) keep old versions of records around in the repository, and allow referencing and accessing multiple versions of the same record.

**Comment by DavidHelfrich42** [comment]
> A valid argument against allowing editing is that a malicious user could post something, wait for comments and reposts, and then alter it to make it look like people were commenting on something else entirely. Perhaps a compromise would be to allow editing the most recent post only, within a fairly short time window, just to deal with those "doh" moments when you spot the typo right after pressing Send. In any case, there should be a flag visible on edited messages and a way to view the original text

I agree with this, but maybe there could be a grace period... like you can edit for the first 5 min then it's locked in.

**Comment by verdverm** [comment]
@RobertSalm I wrote some thoughts on editing and a method we can use today outside of the Bluesky app
https://verdverm.com/blog/adding-record-editing-with-history-to-atprotocol

There are certainly various social and philo aspects, which I touch on, but also with ATProto, we can experiment and have competition to see what does and doesn't work or resound with users.
